docs/index.md

```diff
@@ -0,0 +1,38 @@
+## Introduction
+
+You can learn from this guide how to use OAuth via a Google server to:
+
+- [connect to Settle](Connecting_to_Settle.md)
+- [accept connections from Settle](Accepting_connections_from_Settle.md)
+
+We are using Google Cloud Platform (GCP) service accounts and
+authorization servers to secure endpoints in our APIs.
+
+We propose that you use the same principles to secure endpoints in our APIs.
+
+After reading the docs for how to connect, consider taking a look at
+[the Python demo for the entire flow](python_demo.md).
+
+## The OAuth flow
+
+This is a generic diagram that applies in both directions, i.e. for both
+connecting to settle and for accepting connections from Settle.
+Please refer back to this for clarity on terminology and discrete steps
+if any confusion arises while reading the other docs.
+
+```mermaid
+sequenceDiagram
+    participant C as Client Service
+    participant G as GCP Auth Server
+    participant S as API Server
+    C ->>+ G: SvcAcctKey, Audience
+    Note right of C: Audience = Server endpoint URL
+    G ->> G: Verifies SvcAcctEmail<br>with Client Key
+    G ->>- C: Token
+    C ->>+ S: Token
+    S ->>+ G: Token
```

Comment:
ISTR using the GCP token service made it complicated to run Cardhouse locally. Any thoughts on this?

(I'm dumping all sorts of random thoughts on these concepts, let me know if it's too rambling.)

Reply:
Anything that runs locally ought to use the developer's credentials instead of the ADC (application default credentials) when running in the cloud.

This implies setting an environment variable, as detailed here, when using `google.oauth2.id_token.fetch_id_token`:
https://google-auth.readthedocs.io/en/master/user-guide.html#identity-tokens

However, in `core`, and most likely other Python 2 services, we have a very old version of the library that doesn't support `fetch_id_token` and a more involved method is used to get the token from the auth server. In that scenario, we need extra code that reads the local `.json` credentials file.

Reply:
Sounds good. I wonder whether we just didn't think of that last year, or there was some other problem. I'll try Cardhouse locally again one day and see.